src/Repository/CarrinhoRepository.php

```diff
@@ -16,12 +17,27 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, Carrinho::class);
     }
 
-    public function salvar(Carrinho $carrinho): Carrinho
+    public function salvar(Carrinho $carrinho): int
     {
         $this->getEntityManager()->persist($carrinho);
         $this->getEntityManager()->flush();
         
-        return $carrinho;
+        return $carrinho->getId();
     }
 
+    public function buscar(Cliente $cliente): ?int
+    {
+        $query = $this->createQueryBuilder('carrinho')
+            ->select('carrinho.id')
+            ->where('carrinho.cliente = :cliente')
+            ->andWhere('carrinho.status != :status')
+            ->setParameter('cliente', $cliente)
+            ->setParameter('status', 'Concluído')
+            ->setMaxResults(1)
+            ->getQuery();
+
+        $resultado = $query->getOneOrNullResult();
+
+        return $resultado ? (int) $resultado['id'] : null;
+    }
```

Comment:
Usa a constante no status, e fiquei em duvida se tu deveria usar o concluido mesmo...

O repository sempre manipula o Carrinho e não as propriedades dele, então só retorna o Carrinho e boas...
se quiser garantir que não vem null já que quando tiver nesse status deve ter um Carrinho, então faz um assert de Carrinho diferente de null e se quebrar alguém fez cagada, pois nesse ponto o Carrinho sempre vai existir...

Acho que vai precisar fazer um order by desc ai para sempre pegar o último registro inserido!

```suggestion
    public function buscarConcluidoPorCliente(Cliente $cliente): ?Carrinho
    {
        return $this->createQueryBuilder('carrinho')
            ->where('carrinho.cliente = :cliente')
            ->andWhere('carrinho.status != :status')
            ->setParameter('cliente', $cliente)
            ->setParameter('status', 'Concluído')
            ->setMaxResults(1)
            ->getQuery()
            ->getOneOrNullResult();
    }
```